go/vt/mysqlctl/backupengine.go

```diff
@@ -34,47 +34,121 @@ import (
 
 var (
 	// BackupEngineImplementation is the implementation to use for BackupEngine
-	backupEngineImplementation = flag.String("backup_engine_implementation", builtin, "which implementation to use for the backup method, builtin or xtrabackup")
+	backupEngineImplementation = flag.String("backup_engine_implementation", builtinBackupEngineName, "Specifies which implementation to use for creating new backups (builtin or xtrabackup). Restores will always be done with whichever engine created a given backup.")
 )
 
-// BackupEngine is the interface to the backup engine
+// BackupEngine is the interface to take a backup with a given engine.
 type BackupEngine interface {
 	ExecuteBackup(ctx context.Context, cnf *Mycnf, mysqld MysqlDaemon, logger logutil.Logger, bh backupstorage.BackupHandle, backupConcurrency int, hookExtraEnv map[string]string) (bool, error)
-	ExecuteRestore(ctx context.Context, cnf *Mycnf, mysqld MysqlDaemon, logger logutil.Logger, dir string, bhs []backupstorage.BackupHandle, restoreConcurrency int, hookExtraEnv map[string]string) (mysql.Position, error)
 }
 
-// BackupEngineMap contains the registered implementations for BackupEngine
-var BackupEngineMap = make(map[string]BackupEngine)
+// RestoreEngine is the interface to restore a backup with a given engine.
+type RestoreEngine interface {
+	ExecuteRestore(ctx context.Context, cnf *Mycnf, mysqld MysqlDaemon, logger logutil.Logger, dir string, bh backupstorage.BackupHandle, restoreConcurrency int, hookExtraEnv map[string]string) (mysql.Position, error)
+}
+
+// BackupRestoreEngine is a combination of BackupEngine and RestoreEngine.
+type BackupRestoreEngine interface {
+	BackupEngine
+	RestoreEngine
+}
 
-// GetBackupEngine returns the current BackupEngine implementation.
-// Should be called after flags have been initialized.
+// BackupEngineMap contains the registered implementations for BackupEngine
+var BackupEngineMap = make(map[string]BackupRestoreEngine)
```

Comment:
I suggest renaming this to BackupRestoreEngineMap to reflect its new purpose. Also fix doc comment.